saltedge-ios-swift/Classes/Extensions/CodableExtensions.swift

```diff
@@ -109,8 +125,91 @@ extension UnkeyedDecodingContainer {
         return array
     }
 
-    mutating func decode(_ type: Dictionary<String, Any>.Type) throws -> Dictionary<String, Any> {
+    mutating func decode(_ type: [String: Any].Type) throws -> [String: Any] {
         let nestedContainer = try self.nestedContainer(keyedBy: JSONCodingKeys.self)
         return try nestedContainer.decode(type)
     }
 }
+
+// MARK: - Encoding Extensions
+extension KeyedEncodingContainer {
+    mutating func encodeIfPresent(_ value: [String: Any]?, forKey key: KeyedEncodingContainer<K>.Key) throws {
+        guard let safeValue = value, !safeValue.isEmpty else {
+            return
+        }
+        var container = self.nestedContainer(keyedBy: JSONCodingKeys.self, forKey: key)
+        for item in safeValue {
+            guard let key = JSONCodingKeys(stringValue: item.key) else {
+                continue
+            }
+
+            if let val = item.value as? Int {
+                try container.encodeIfPresent(val, forKey: key)
+            } else if let val = item.value as? String {
+                try container.encodeIfPresent(val, forKey: key)
+            } else if let val = item.value as? Double {
+                try container.encodeIfPresent(val, forKey: key)
+            } else if let val = item.value as? Float {
+                try container.encodeIfPresent(val, forKey: key)
+            } else if let val = item.value as? Bool {
+                try container.encodeIfPresent(val, forKey: key)
+            } else if let val = item.value as? [Any] {
+                try container.encodeIfPresent(val, forKey: key)
+            } else if let val = item.value as? [String: Any] {
+                try container.encodeIfPresent(val, forKey: key)
+            }
+        }
+    }
+
+    mutating func encodeIfPresent(_ value: [Any]?, forKey key: KeyedEncodingContainer<K>.Key) throws {
+        guard let safeValue = value else {
+            return
+        }
+        if let val = safeValue as? [Int] {
+            try self.encodeIfPresent(val, forKey: key)
+        } else if let val = safeValue as? [String] {
+            try self.encodeIfPresent(val, forKey: key)
+        } else if let val = safeValue as? [Double] {
+            try self.encodeIfPresent(val, forKey: key)
+        } else if let val = safeValue as? [Float] {
+            try self.encodeIfPresent(val, forKey: key)
+        } else if let val = safeValue as? [Bool] {
+            try self.encodeIfPresent(val, forKey: key)
+        } else if let val = value as? [[String: Any]] {
+            var container = self.nestedUnkeyedContainer(forKey: key)
+            try container.encode(contentsOf: val)
+        }
+    }
+}
+
+extension UnkeyedEncodingContainer {
+    mutating func encode(contentsOf sequence: [[String: Any]]) throws {
+        for dict in sequence {
+            try self.encodeIfPresent(dict)
+        }
+    }
+
+    mutating func encodeIfPresent(_ value: [String: Any]) throws {
```

Comment:
I would suggest replacing it with **switch**.
 This code does the same thing as the original one, but is more compact and readable thanks to the use of the **switch** operator.